Services/AbstractExporter.php

```diff
@@ -69,18 +69,20 @@ public function export(array $criteria = [], array $options = []): void
     /**
      * Get datetime based on user's date format.
      *
-     * @return \Carbon\Carbon|null
+     * @return \Carbon\CarbonImmutable|null
      */
-    public function getDateTime(string $value = null, string $default = null): ?Carbon
+    public function getDateTime(string $value = null, string $default = null): ?CarbonImmutable
     {
         if (null !== $value) {
             if ($format = ($_SESSION['usersettings.language.date_format'] ?? null)) {
                 // See https://www.php.net/manual/en/datetimeimmutable.createfromformat.php for details on `!`.
-                return Carbon::createFromFormat('!' . $format, $value) ?: new Carbon($default);
+                $returnDate = CarbonImmutable::createFromFormat('!' . $format, $value) ?: new CarbonImmutable($default);
+                /* @phpstan-ignore-next-line */
+                return $this->dateTimeHelper->parseUserDateTime($value)->setToDbTimezone();
             }
         }
 
-        return new Carbon($default);
+        return new CarbonImmutable($default);
```

Comment:
`$default` should be handled just like `$value`, i.e. passed through the date time helper functions, and then the whole function body can (probably) be rewritten to something like

```php
$value ??= $default;

return null !== $value ? $this->dateTimeHelper->parseUserDateTime($value)->setToDbTimezone() : null;
```